ch_pipeline/core/dataquery.py

```diff
@@ -655,6 +658,111 @@ def next(self, ts):
         return inputs
 
 
+class QueryFrequencyMap(task.MPILoggedTask):
+    """Get the CHIME frequency map that was active when this data was collected.
+
+    Attributes
+    ----------
+    cache : bool
+        Only query for the inputs for the first container received. For all
+        subsequent files just return the initial set of inputs. This can help
+        minimise the number of potentially fragile database operations.
+    """
+
+    cache = config.Property(proptype=bool, default=False)
+
+    _cached_inputs = None
+
+    def next(self, ts) -> FrequencyMapSingle:
+        """Generate an input description from the timestream passed in.
+
+        Parameters
+        ----------
+        ts : andata.CorrData
+            Timestream container.
+
+        Returns
+        -------
+        freq_map : FrequencySlotMap
+            Frequency slot map container
+        """
+
+        # Fetch from the cache if we can
+        if self.cache and self._cached_inputs:
+            self.log.debug("Using cached inputs.")
+            return self._cached_inputs
+
+        stream_id = None
+        stream = None
+
+        if mpiutil.rank0:
+            layout.connect_database()
+
+            # Check to see if this timestream has a dataset id, and assign it
+            # if it does. If there is no dataset_id, we will just assume a
+            # default mapping
+            if (did_flag := ts.flags.get("dataset_id", None)) is not None:
+                # Cast bytearry type to unicode if needed
+                did_flag = memh5.ensure_unicode(did_flag[:])
+                # Get only unique dataset ids to reduce number of lookups. Ignore
+                # the null dataset at the 0th index
+                unique_dataset_ids = np.unique(did_flag)[1:]
+
+                if len(unique_dataset_ids) != 0:
+                    fmaps = {}
+                    # Cast each entry to a python string due to a bug with mySQL
+                    # typ conversion
+                    unique_dataset_ids = [str(dsid) for dsid in unique_dataset_ids]
```

Comment:
You might be able to use the routines here (https://github.com/chime-experiment/chimedb_dataset/blob/master/chimedb/dataset/utils.py#L76) to simplify this. Have a look, but I might not be worth it.

Reply:
Made some tweaks based on this, but can't call it directly since I also need `state.data` and would have to do that lookup anyway